pyproject.toml

```
@@ -1,32 +1,126 @@
 [build-system]
-requires = ["jupyter_packaging~=0.10", "jupyterlab>=4.0.0a25,<5", "pre-commit"]
-build-backend = "jupyter_packaging.build_api"
+requires = ["hatchling>=1.0", "jupyterlab>=4.0.0a25,<5"]
+build-backend = "hatchling.build"
 
-[license]
-file="LICENSE"
+[project]
+name = "notebook"
+description = "Jupyter Notebook - A web-based notebook environment for interactive computing"
+readme = "README.md"
+license = { file = "LICENSE" }
+requires-python = ">=3.7"
+authors = [
+    { name = "Jupyter Development Team", email = "jupyter@googlegroups.com" },
+]
+keywords = [
+    "Jupyter",
+    "JupyterLab",
+    "Notebook",
+]
+classifiers = [
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+]
+dependencies = [
+    "jupyter_server>=1.16.0,<2",
+    "jupyterlab>=4.0.0a24,<5",
+    "jupyterlab_server>=2.13,<3",
+    "notebook_shim>=0.1,<0.2",
+    "tornado>=6.1.0",
+]
+dynamic = ["version"]
 
-[tool.jupyter-packaging.options]
-skip-if-exists = ["notebook/labextension/static/style.js", "notebook/static/bundle.js"]
-ensured-targets = ["notebook/labextension/static/style.js", "notebook/static/bundle.js"]
+[project.scripts]
+jupyter-notebook = "notebook.app:main"
 
-[tool.jupyter-packaging.builder]
-factory = "jupyter_packaging.npm_builder"
+[project.urls]
+Documentation = "https://jupyter-notebook.readthedocs.io/"
+Homepage = "https://github.com/jupyter/notebook"
+Source = "https://github.com/jupyter/notebook"
+Tracker = "https://github.com/jupyter/notebook/issues"
 
-[tool.jupyter-packaging.build-args]
-build_cmd = "build:prod"
-npm = ["jlpm"]
+[project.optional-dependencies]
+test = [
+    "coverage",
+    "nbval",
+    "pytest>=6.0",
+    "pytest-cov",
+    "requests",
+    "pytest-tornasync",
+    "pytest-timeout",
+    "pytest-console-scripts",
+    "ipykernel",
+    "jupyterlab_server[test]>=2.13,<3",
+]
+dev = [
+    "pre-commit",
+    "bump2version",
+    "hatchling"
+]
+
+[tool.hatch.version]
+path = "notebook/_version.py"
+source = "code"
+
+[tool.hatch.build.targets.wheel.shared-data]
+"notebook/labextension" = "share/jupyter/labextensions/@jupyter-notebook/lab-extension"
```

@blink1073 I'm not an expert on the filesystem layout for JupyterLab extensions, but should this be
```suggestion
"notebook/labextension" = "share/jupyter/labextensions/@jupyter-notebook"
```

---

Yes, I agree

---

The extension is nested in our case, under `@jupyter-notebook`.   The full extension name is `@jupyter-notebook/lab-extension`.

---

Right, I'm with you. This is where the fact that `/` is allowed in NPM package names threw me for a second!